AxonDeepSeg/train_network.py

```diff
@@ -75,63 +74,108 @@ def train_model(path_trainingset, path_model, config, path_model_init=None,
+    #If the key values of augmentation are set to True then their respective probability are set to 0.5 else to 0. Probalility(p) suggests a certainity of applying data augmentation operations (shift, rotate, blur, elastic, flip) to an image.
+
+    # Probability value of 0.5 is chosen so that the original as well augmented image are taken into account while training the model.
+    if shifting:
+        p_shift = 0.5
+    if rotation:
+        p_rotate = 0.5
+    if flipping:
+        p_flip = 0.5
+    if gaussian_blur:
+        p_blur = 0.5
+    if elastic:
+        p_elastic = 0.5
+
+
+    #####Data Augmentation parameters#####
+
+    #Elastic transform parameters
+    alpha_max = 9
+    sigma = 3
+    alpha = random.choice(list(range(1, alpha_max)))
+
+    #Random rotation parameters
+    low_bound = 5
+    high_bound = 89
+
+    #Shifting parameters
+    percentage_max = 0.1
+    size_shift = int(percentage_max*image_size)
+    low_limit = 0
+    high_limit = (2*size_shift-1) / image_size
+
 
-    #####################Training###################
-    image_datagen = ImageDataGenerator(**data_gen_args)
-    mask_datagen = ImageDataGenerator(**data_gen_args)
+    ######################################
 
-    seed = 2018
 
-    # Image and Mask Data Generator
-    image_generator_train = image_datagen.flow(image_train_gen, y=None, seed = seed,  batch_size=batch_size, shuffle=True)
-    mask_generator_train = mask_datagen.flow(mask_train_gen, y=None, seed = seed, batch_size=batch_size, shuffle=True)
+    AUGMENTATIONS_TRAIN = Compose([
 
-    # Just zip the two generators to get a generator that provides augmented images and masks at the same time
-    train_generator = zip(image_generator_train, mask_generator_train)
+    #Randomy flips an image either horizontally, vertically or both.
+    Flip(p = p_flip),
 
-    ###########################Validation###########
-    data_gen_args = dict()
-    image_datagen = ImageDataGenerator(**data_gen_args)
-    mask_datagen = ImageDataGenerator(**data_gen_args)
+    #Randomly rotates an image between low limit and high limit.
+    ShiftScaleRotate(shift_limit=(low_limit, high_limit) , scale_limit=(0,0),rotate_limit=(0,0), border_mode=border_mode, p = p_shift, interpolation = cv2.INTER_NEAREST),
 
+    #Randomly applies elastic transformation on the image.
+    ElasticTransform(alpha= alpha, sigma= sigma, p = p_elastic, alpha_affine = alpha),
 
-    image_generator_valid = image_datagen.flow(x=image_valid_gen, y=None,  batch_size=batch_size, seed = seed, shuffle=False)
-    mask_generator_valid = mask_datagen.flow(x=mask_valid_gen, y=None,  batch_size=batch_size, seed = seed, shuffle=False)
+    #Blurs an image using gaussian kernal.
+    GaussianBlur(p = p_blur),
+
+    #Randomly rotates the image between low bound and high bound.
+    Rotate(limit=(low_bound, high_bound), border_mode=border_mode, p = p_rotate, interpolation = cv2.INTER_NEAREST)
+
+    ])
+
```

Comment:
any reason to keep all those empty lines?

Reply:
Just because  there are many data augmentation operations done on the image, and to make it easy to understand, I have made some comments on the top of each operation while keeping some empty lines.

Reply:
>and to make it easy to understand, I have made some comments on the top of each operation while keeping some empty lines.

The best way to make code easy to read is by following the PEP8 standard.

I'll do a pass of the new files in this branch using [black](https://github.com/psf/black) to apply these standards automatically.

Reply:
Resolved in 9ebed10